apps/site/assets/ts/models/alert.ts

```diff
@@ -19,6 +20,93 @@ export const alertsByStop = (alerts: Alert[], stopId: StopId): Alert[] =>
       !!entities.stop && entities.stop!.some((id: StopId) => id === stopId)
   );
 
+const hasEffect = (alerts: Alert[], effect: string): boolean =>
+  some(alerts, alert => alert.effect === effect);
+
+export const isSuspension = (alerts: Alert[]): boolean =>
+  hasEffect(alerts, "suspension");
+
+export const isShuttleService = (alerts: Alert[]): boolean =>
+  hasEffect(alerts, "shuttle");
+
+export const isDetour = (alerts: Alert[]): boolean =>
+  hasEffect(alerts, "detour");
+
+export const alertsWithStop = (alerts: Alert[]): Alert[] =>
+  alerts.filter(
+    ({ informed_entity: entites }: Alert): boolean => !!entites.stop
+  );
+
+export const routeWideAlerts = (alerts: Alert[]): Alert[] =>
+  alerts.filter(({ informed_entity: { entities } }: Alert): boolean =>
+    entities.some(entity => !entity.stop && !entity.trip)
+  );
+
+// Can there be alerts that don't affect routes
+export const alertsByRoute = (alerts: Alert[]): { [key: string]: Alert[] } => {
+  return reduce(
+    alerts,
+    (acc, alert) => {
+      // create a map of every route id to the alert
+      const explodedAlerts = reduce(
+        alert.informed_entity.route,
+        (innerAcc, routeId) => {
+          return { ...innerAcc, [routeId]: [alert] };
+        },
+        {}
+      );
+
+      return mergeWith(acc, explodedAlerts, (objValue, srcValue) => {
+        if (isArray(objValue)) {
+          return objValue.concat(srcValue);
+        }
+        return srcValue;
+      });
+    },
+    {}
+  );
+};
+
+// This will only return alerts with direction id specified (aka affecting a single direction)
```

Comment:
question: Does null direction id imply that an alert affects both directions? If so, why would we want to exclude those alerts?

Reply:
This was broken out because `obj[null]` isn't allowed by typescript.
`alertsByDirectionId` would map alerts affecting both directions to the key of "null".
The only way to access it would be to use `alertsByDirectionObj["null"]` (instead of `alertsByDirectionObj[null]`).
Typescript discourages this as it seems error prone. I broke it out into its own function instead.

Reply:
Agree that it seems error prone to have a [null] key. 
Suggestion (non-blocking): When directionId is null, I wonder if it would simplify things to add it to the list *both* directionIds so that alertsByDirectionId returns a full picture and a separate function isn't needed.